### PR TITLE
feat(tldr): update to latest version

### LINKED ISSUE #1826: tldr - Oudated spec.
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [X] I would like to work on this.

### What CLI tool does this relate to?

tldr

### Which statement makes the most sense?

There is a bug with this completion spec

### Issue Details

The `tldr` spec is outdated.

1. The local cache folder needs to be corrected. It is moved from `~/.tldr/cache` to `~/.tldrc/tldr`.
2. `-p, --platform` option is missing.
3. The following options are no longer supported.
   * `-a, --list-all`
   * `--random`
   * `--random-example`
   * `-m, --markdown`

### PR DESCRIPTION
**Pull request by seenukarthi**
This PR fixes the completion spec `tldr`
 
Closes #1826

Sgned-off-by: Karthikeyan Vaithilingam <seenukarthi@gmail.com>

### CODE REVIEW

File: src/tldr.ts [hunk 1]
```diff
@@ -139,6 +123,23 @@ const completionSpec: Fig.Spec = {
...
+      description: "Select platform",
+      args: {
+        name: "platform",
+        suggestions: ["linux", "osx", "sunos", "windows", "common"],
+      },
+    },
+    {
+      name: "--platform",
+      description: "Select platform",
+      insertValue: "--platform={cursor}",
```
Comment (AlanSyue) [inline src/tldr.ts:137]: Maybe we can use `requiresSeparator` instead.
```suggestion
      requiresSeparator: true,
```
https://github.com/withfig/autocomplete-tools/blob/main/types/index.d.ts#L713-L725

File: src/tldr.ts [hunk 2]
```diff
@@ -139,6 +123,23 @@ const completionSpec: Fig.Spec = {
       name: ["-c", "--clear-cache"],
       description: "Delete the entire local cache",
     },
+    {
+      name: "-p",
+      description: "Select platform",
+      args: {
+        name: "platform",
+        suggestions: ["linux", "osx", "sunos", "windows", "common"],
```
Comment (AlanSyue) [inline src/tldr.ts:131]: [nit]
Maybe we should declare a variable to gather these keywords?

https://github.com/withfig/autocomplete/pull/1827/files#diff-50bd02ab6f158b81b333c17f0cc4c4058f28dbdb7e710d9d0acb4cca77a99a13R140

**Review by AlanSyue — APPROVED** [review]

File: src/tldr.ts (hunk 2 above)
Comment (seenukarthi) [reply src/tldr.ts]: @AlanSyue Yes have made the suggestions constant.

**Review by AlanSyue — APPROVED** [review]

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks @seenukarthi! I left a comment to solve before merging this.

File: src/tldr.ts [hunk 3]
```diff
@@ -139,6 +125,23 @@ const completionSpec: Fig.Spec = {
...
+    },
+    {
+      name: "--platform",
+      description: "Select platform",
+      requiresSeparator: true,
+      args: {
+        name: "platform",
+        suggestions: platformSuggestions,
+      },
+    },
```
[the remark this answers is not in the document]
Comment (seenukarthi) [reply src/tldr.ts:144]: @fedeci the option `-p` expects a white space between `-p` and the argument, on the other hand `--platform` need a `=` between option and the argument that's the reason I made it as desperate options.

[the remark this answers is not in the document]
Comment (fedeci) [reply src/tldr.ts:144]: `--platform` is not requiring the equals nor `-p` requires a space.

**Pull request merged**